extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/State.java

```diff
@@ -67,18 +83,20 @@ public boolean equals(Object o) {
         }
 
         State state = (State) o;
-        return partitionsToOffset.equals(state.partitionsToOffset);
+        return Objects.equals(taskConfigs, state.taskConfigs) &&
+               Objects.equals(partitionsToOffset, state.partitionsToOffset);
     }
 
     @Override
     public int hashCode() {
-        return Objects.hash(partitionsToOffset);
+        return Objects.hash(taskConfigs, partitionsToOffset);
     }
 
     @Override
     public String toString() {
         return "State{" +
-                "partitionsToOffset=" + partitionsToOffset +
-                '}';
+               "taskConfigs=" + taskConfigs +
```

Comment:
you can reveal credentials by doing so

Reply:
TaskConfig is List<Map<String, String>>. It contains offsets, I don't think it credentials.